grafana-operator/grafana-operator
# Migration from 3.x.x to 4.0.0

## Description

Changes required to release 4.0.0, and make it work via OLM. This should also contain the migration guide from 3.x.x to 4.0.0.

CRDs are upgraded to v1 (from v1alpha1).

## TODO

1. Automatically upgrade CRs to v1 (see https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#upgrade-existing-objects-to-a-new-stored-version)
2. Finish migration documentation

## Verification steps

1. Install 3.10.3 via OLM
2. Deploy Grafana and some Dashboards
3. Uninstall 3.10.3 via OLM
4. Apply the CatalogSource (provided below)
5. Install 4.0.0
6. Verify that all imported Dashboards are still present.

The following CatalogSource can be used to install 4.0.0 via OLM:

```yaml
apiVersion: operators.coreos.com/v1alpha1
kind: CatalogSource
metadata:
  name: grafana-operator-manifests
  namespace: grafana
spec:
  sourceType: grpc
  image: quay.io/grafana-operator/controller-index:v4.0.0
```
